client/src/components/signup.js

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from "react";
+import "./signup.css"
+import loginavi from "../loginavi.png";
+import { Link, useNavigate } from "react-router-dom";
+
+const SignUp = () => {
+    const [username, setUsername] = useState('')
+    const [password, setPassword] = useState('')
+    const [email, setEmail] = useState('')
+    const [passwordConfirmation, setPasswordConfirmation] = useState('')
+    const [profilePhoto, setProfilePhoto] = useState('')
+    const [isLoading, setIsLoading] = useState(false)
+    const navigate = useNavigate()
+    const [errors, setErrors] = useState([]);
+
+    const handleSubmit = (e) => {
+      e.preventDefault();
+      setIsLoading(true);
+      setErrors([]);
+      fetch('/signup', {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json"
+        },
+        body: JSON.stringify({
+          username,
```

Comment:
I think this needs to be an object, like `fetch('/signup', {
method: ...,
headers: {....},
body: JSON.stringify({username: username, 
password: password...})})`